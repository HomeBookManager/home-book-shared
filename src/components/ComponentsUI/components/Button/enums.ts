export enum ButtonColor {
  primary = 'primary',
  secondary = 'secondary',
  success = 'success',
  warning = 'warning',
  error = 'error',
}

export enum ButtonSize {
  small = 'small',
  medium = 'medium',
  large = 'large',
}

export enum ButtonVariant {
  text = 'text',
  contained = 'contained',
  outlined = 'outlined',
}
