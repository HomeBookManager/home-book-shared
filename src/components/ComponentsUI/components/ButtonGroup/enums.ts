export enum ButtonGroupOrientation {
  horizontal = 'horizontal',
  vertical = 'vertical',
}
