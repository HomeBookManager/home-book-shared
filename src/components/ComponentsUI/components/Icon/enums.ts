export enum IconShape {
  downOutlined = 'downOutlined',
  leftOutlined = 'leftOutlined',
  rightOutlined = 'rightOutlined',
  upOutlined = 'upOutlined',
}
