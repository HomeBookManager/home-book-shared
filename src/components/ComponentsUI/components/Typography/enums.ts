export enum TypographyFontStyle {
  italic = 'italic',
  normal = 'normal',
}

export enum TypographyFontType {
  h1 = 'h1',
  h2 = 'h2',
  h3 = 'h3',
  h4 = 'h4',
  h5 = 'h5',
  h6 = 'h6',
  p = 'p',
  small = 'small',
}

export enum TypographyFontWeight {
  black = 'black',
  bold = 'bold',
  extraBold = 'extra-bold',
  light = 'light',
  medium = 'medium',
  regular = 'regular',
  semiBold = 'semi-bold',
}
