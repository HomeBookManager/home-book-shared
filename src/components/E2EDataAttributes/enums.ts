export enum E2EAttribute {
  button = 'button',
  icon = 'icon',
  test = 'test',
  text = 'text',
}
