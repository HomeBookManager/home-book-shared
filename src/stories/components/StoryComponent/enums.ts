export enum ContentAlignItems {
  top = 'top',
  center = 'center',
  bottom = 'bottom',
}

export enum ContentDisplay {
  block = 'block',
  flex = 'flex',
  grid = 'grid',
}

export enum ContentGridFlow {
  column = 'column',
  row = 'row',
  maxTwoColumns = 'max-two-columns',
  maxThreeColumns = 'max-three-columns',
  maxFourColumns = 'max-four-columns',
  maxFiveColumns = 'max-five-columns',
  maxSixColumns = 'max-six-columns',
}
