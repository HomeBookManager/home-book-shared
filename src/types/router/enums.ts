import { Action } from 'history';

export enum RouterAction {
  push = Action.Push,
  replace = Action.Replace,
}
