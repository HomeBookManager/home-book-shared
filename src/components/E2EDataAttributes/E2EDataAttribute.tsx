import React, { Children, FC, ReactElement } from 'react';

// types
import { E2EAttribute } from './enums';
import { E2EValue } from './types';

// utils
import { getDataTestAttribute } from './utils';

export type TProps = {
  children: ReactElement<HTMLElement>;
  type?: E2EAttribute | string;
  value?: E2EValue;
};

const E2EDataAttribute: FC<TProps> = ({ children, type = '', value = '' }) => {
  const childrenOnly = Children.only(children);

  if (!type && !value) {
    return childrenOnly;
  }

  return React.cloneElement(childrenOnly, {
    [getDataTestAttribute(type)]: value,
  });
};

export default E2EDataAttribute;
