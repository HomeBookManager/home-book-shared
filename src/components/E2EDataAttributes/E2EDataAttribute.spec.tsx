import { render } from '@testing-library/react';

// components
import E2EDataAttribute from './E2EDataAttribute';

// types
import { E2EAttribute } from './enums';

describe('E2EDataAttribute snapshots', () => {
  it('should render E2EDataAttribute', () => {
    // before
    const { asFragment } = render(
      <E2EDataAttribute type={E2EAttribute.test}>
        <span>child</span>
      </E2EDataAttribute>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with value', () => {
    // before
    const { asFragment } = render(
      <E2EDataAttribute type={E2EAttribute.test} value="test">
        <span>child</span>
      </E2EDataAttribute>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render without attribute id', () => {
    // before
    const { asFragment } = render(
      <E2EDataAttribute>
        <span>child</span>
      </E2EDataAttribute>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
