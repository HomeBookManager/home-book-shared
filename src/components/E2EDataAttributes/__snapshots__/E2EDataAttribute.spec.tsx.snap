// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2EDataAttribute snapshots should render E2EDataAttribute 1`] = `
<DocumentFragment>
  <span
    data-test-test=""
  >
    child
  </span>
</DocumentFragment>
`;

exports[`E2EDataAttribute snapshots should render with value 1`] = `
<DocumentFragment>
  <span
    data-test-test="test"
  >
    child
  </span>
</DocumentFragment>
`;

exports[`E2EDataAttribute snapshots should render without attribute id 1`] = `
<DocumentFragment>
  <span>
    child
  </span>
</DocumentFragment>
`;
